File: backend/src/main/java/com/lawfirm/domain/hr/entity/Employee.java
package com.lawfirm.domain.hr.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.lawfirm.common.base.BaseEntity;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.time.LocalDate;

/**
 * 员工档案实体
 */
@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@TableName("hr_employee")
public class Employee extends BaseEntity {

    /**
     * 关联用户ID
     */
    private Long userId;

    /**
     * 工号
     */
    private String employeeNo;

    /**
     * 性别：MALE-男, FEMALE-女
     */
    private String gender;

    /**
     * 出生日期
     */
    private LocalDate birthDate;

    /**
     * 身份证号
     */
    private String idCard;

    /**
     * 国籍
     */
    private String nationality;

    /**
     * 籍贯
     */
    private String nativePlace;

    /**
     * 政治面貌
     */
    private String politicalStatus;

    /**
     * 学历
     */
    private String education;

    /**
     * 专业
     */
    private String major;

    /**
     * 毕业院校
     */
    private String graduationSchool;

    /**
     * 毕业日期
     */
    private LocalDate graduationDate;

    /**
     * 紧急联系人
     */
    private String emergencyContact;

    /**
     * 紧急联系电话
     */
    private String emergencyPhone;

    /**
     * 家庭住址
     */
    private String address;

    /**
     * 律师执业证号
     */
    private String lawyerLicenseNo;

    /**
     * 执业证发证日期
     */
    private LocalDate licenseIssueDate;

    /**
     * 执业证到期日期
     */
    private LocalDate licenseExpireDate;

    /**
     * 执业证状态：VALID-有效, EXPIRED-已过期, SUSPENDED-已暂停
     */
    private String licenseStatus;

    /**
     * 执业领域
     */
    private String practiceArea;

    /**
     * 执业年限
     */
    private Integer practiceYears;

    /**
     * 职位
     */
    private String position;

    /**
     * 职级
     */
    private String level;

    /**
     * 入职日期
     */
    private LocalDate entryDate;

    /**
     * 试用期结束日期
     */
    private LocalDate probationEndDate;

    /**
     * 转正日期
     */
    private LocalDate regularDate;

    /**
     * 离职日期
     */
    private LocalDate resignationDate;

    /**
     * 离职原因
     */
    private String resignationReason;

    /**
     * 工作状态：ACTIVE-在职, PROBATION-试用, RESIGNED-离职, RETIRED-退休
     */
    private String workStatus;

    /**
     * 备注
     */
    private String remark;
}

File: frontend/apps/web-antd/src/api/hr/employee.ts
import type { PageResult } from '../matter/types';

/**
 * 员工管理 API
 */
import { requestClient } from '#/api/request';

// ========== 类型定义 ==========
export interface EmployeeDTO {
  id: number;
  userId?: number;
  employeeNo?: string;
  realName?: string;
  email?: string;
  phone?: string;
  departmentId?: number;
  departmentName?: string;
  gender?: string;
  birthDate?: string;
  idCard?: string;
  nationality?: string;
  nativePlace?: string;
  politicalStatus?: string;
  education?: string;
  major?: string;
  graduationSchool?: string;
  graduationDate?: string;
  emergencyContact?: string;
  emergencyPhone?: string;
  address?: string;
  lawyerLicenseNo?: string;
  licenseIssueDate?: string;
  licenseExpireDate?: string;
  licenseStatus?: string;
  practiceArea?: string;
  practiceYears?: number;
  position?: string;
  level?: string;
  entryDate?: string;
  probationEndDate?: string;
  regularDate?: string;
  resignationDate?: string;
  resignationReason?: string;
  workStatus?: string;
  workStatusName?: string;
  remark?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface EmployeeQuery {
  pageNum?: number;
  pageSize?: number;
  employeeNo?: string;
  realName?: string;
  departmentId?: number;
  workStatus?: string;
  position?: string;
}

export interface CreateEmployeeCommand {
  userId: number;
  employeeNo?: string;
  gender?: string;
  birthDate?: string;
  idCard?: string;
  nationality?: string;
  nativePlace?: string;
  politicalStatus?: string;
  education?: string;
  major?: string;
  graduationSchool?: string;
  graduationDate?: string;
  emergencyContact?: string;
  emergencyPhone?: string;
  address?: string;
  lawyerLicenseNo?: string;
  licenseIssueDate?: string;
  licenseExpireDate?: string;
  licenseStatus?: string;
  practiceArea?: string;
  practiceYears?: number;
  position?: string;
  level?: string;
  entryDate?: string;
  probationEndDate?: string;
  workStatus?: string;
  remark?: string;
}

export interface UpdateEmployeeCommand {
  employeeNo?: string;
  gender?: string;
  birthDate?: string;
  idCard?: string;
  nationality?: string;
  nativePlace?: string;
  politicalStatus?: string;
  education?: string;
  major?: string;
  graduationSchool?: string;
  graduationDate?: string;
  emergencyContact?: string;
  emergencyPhone?: string;
  address?: string;
  lawyerLicenseNo?: string;
  licenseIssueDate?: string;
  licenseExpireDate?: string;
  licenseStatus?: string;
  practiceArea?: string;
  practiceYears?: number;
  position?: string;
  level?: string;
  entryDate?: string;
  probationEndDate?: string;
  regularDate?: string;
  workStatus?: string;
  remark?: string;
}

// ========== API 函数 ==========

/** 分页查询员工档案 */
export function getEmployeeList(params: EmployeeQuery) {
  return requestClient.get<PageResult<EmployeeDTO>>('/hr/employee', { params });
}

/** 根据ID查询员工档案 */
export function getEmployeeDetail(id: number) {
  return requestClient.get<EmployeeDTO>(`/hr/employee/${id}`);
}

/** 根据用户ID查询员工档案 */
export function getEmployeeByUserId(userId: number) {
  return requestClient.get<EmployeeDTO>(`/hr/employee/user/${userId}`);
}

/** 创建员工档案 */
export function createEmployee(data: CreateEmployeeCommand) {
  return requestClient.post<EmployeeDTO>('/hr/employee', data);
}

/** 更新员工档案 */
export function updateEmployee(id: number, data: UpdateEmployeeCommand) {
  return requestClient.put<EmployeeDTO>(`/hr/employee/${id}`, data);
}

/** 删除员工档案 */
export function deleteEmployee(id: number) {
  return requestClient.delete(`/hr/employee/${id}`);
}
